addons/survey/static/src/js/survey_result.js

```diff
@@ -186,95 +267,136 @@ if(!$('.js_surveyresult').length) {
-    }
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
 
-    // Script for filter
-    $('td.survey_answer').hover(function(){
-        $(this).find('i.fa-filter').removeClass('invisible');
-    }, function(){
-        $(this).find('i.fa-filter').addClass('invisible');
-    });
-    $('td.survey_answer i.fa-filter').click(function(){
-        var cell = $(this);
-        var row_id = cell.attr('data-row_id') | 0;
-        var answer_id = cell.attr('data-answer_id');
-        if(document.URL.indexOf("?") == -1){
-            window.location.href = document.URL + '?' + encodeURI(row_id + ',' + answer_id);
-        }
-        else {
-            window.location.href = document.URL + '&' + encodeURI(row_id + ',' + answer_id);
-        }
-    });
-
-    // for clear all filters
-    $('.clear_survey_filter').click(function(){
-        window.location.href = document.URL.substring(0,document.URL.indexOf("?"));
-    });
-    $('span.filter-all').click(function(){
-        event.preventDefault();
-        if(document.URL.indexOf("finished") != -1){
-            window.location.href = document.URL.replace('?finished&','?').replace('&finished&','&').replace('?finished','').replace('&finished','');
-        }
-    }).hover(function(){
-        if(document.URL.indexOf("finished") == -1){
-            $(this)[0].style.cursor = 'default';
-        }
-    });
-    // toggle finished/all surveys filter
-    $('span.filter-finished').click(function(){
-        event.preventDefault();
-        if(document.URL.indexOf("?") == -1){
-            window.location.href = document.URL + '?' + encodeURI('finished');
-        }
-        else if(document.URL.indexOf("finished") == -1){
-            window.location.href = document.URL + '&' + encodeURI('finished');
-        }
-    }).hover(function(){
-        if(document.URL.indexOf("finished") != -1){
-            $(this)[0].style.cursor = 'default';
-        }
-    });
+    /**
+    * @override
+    */
+    willStart: function () {
+        var url = '/web/webclient/locale/' + (document.documentElement.getAttribute('lang') || 'en_US').replace('-', '_');
+        var localeReady = ajax.loadJS(url);
+        return Promise.all([this._super.apply(this, arguments), localeReady]);
+    },
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$('.pagination').each(function (){
+                var questionId = $(this).data("question_id");
+                new publicWidget.registry.SurveyResultPagination(self, {
+                    'questionsEl': self.$('#table_question_'+ questionId)
+                }).attachTo($(this));
+                $(this).find('li:first').addClass('active').find('a').click();
+            });
```

Comment:
attachTo being asynchronous, this click might happen before the attachTo finishes and nothing would happen.

I think it would be a good idea to let the start of the paginationWidget handle the initial swith to the first page with a call to a private method _changePage(pageNumber). You could move most of the logic inside _onPageClick to this new private function and _onPageClick would then call it after extracting the desired page number.

That way we will know for sure that the page will be initially set to the first one and, also, the parent won't have to indirectly command an action to its child.

What do you think ?